src/containers/layouts/messaging/chat2/chat2.component.tsx

```diff
@@ -0,0 +1,222 @@
+
+interface ComponentProps {
+  conversation: ConversationModel;
+  fileSectionOpened: boolean;
+  newMessage: string;
+  galleryFiles: MediaLibrary.Asset[];
+  fileMessageAppearance: ChatFileMessageAppearance; // just for demo purposes
+  onNewMessageChange: (text: string) => void;
+  onMessageAdd: () => void;
+  onAddButtonPress: () => void;
+  onCancelButtonPress: () => void;
+  onPhotoOrVideoButtonPress: () => void;
+  onFileButtonPress: () => void;
+  onLocationButtonPress: () => void;
+  onContactButtonPress: () => void;
+  onTakePhotoButtonPress: () => void;
+  onGalleryItemPress: (item: MediaLibrary.Asset) => void;
+}
+
+export type Chat2ComponentProps = ThemedComponentProps & ComponentProps;
+
+class Chat2Component extends React.Component<Chat2ComponentProps> {
+
+  private listRef: React.RefObject<FlatList<any>> = React.createRef();
+
+  private onNewMessageChange = (text: string): void => {
+    this.props.onNewMessageChange(text);
+  };
+
+  private onMessageAdd = (): void => {
+    this.props.onMessageAdd();
+    this.listRef.current.scrollToEnd();
+  };
+
+  private onAddButtonPress = (): void => {
+    this.props.onAddButtonPress();
+  };
+
+  private createUiMessages = (): UiMessageModel[] => {
+    const { conversation } = this.props;
+
+    return conversation.messages.map((message: MessageModel) => {
+      if (message.author === profile1) {
+        return { ...message, alignment: ChatMessageAlignment.left };
+      } else if (message.author === profile2) {
+        return { ...message, alignment: ChatMessageAlignment.right };
+      }
+    });
+  };
+
+  private renderMessage = (info: ListRenderItemInfo<UiMessageModel>)
+    : React.ReactElement<ChatMessageProps | ChatFileMessageProps> => {
+
+    const { themedStyle, fileMessageAppearance } = this.props;
+
+    return info.item.text ? (
+      <ChatMessage
+        style={themedStyle.message}
+        index={info.index}
+        message={info.item}
+        alignment={info.item.alignment}
+      />
+    ) : (
+      <ChatFileMessage
+        style={themedStyle.message}
+        index={info.index}
+        message={info.item}
+        alignment={info.item.alignment}
+        appearance={fileMessageAppearance}
+      />
+    );
+  };
+
+  private renderSendMessageButton = (): React.ReactElement<ButtonProps> | null => {
+    const { newMessage } = this.props;
+
+    return (newMessage && newMessage.length !== 0) ? (
+      <Button
+        appearance='ghost'
+        size='large'
+        icon={PaperPlaneIcon}
+        onPress={this.onMessageAdd}
+      />
+    ) : null;
+  };
+
+  private renderFileSection = (): React.ReactElement<ViewProps> | null => {
+    const {
+      fileSectionOpened,
+      galleryFiles,
+      onCancelButtonPress,
+      onPhotoOrVideoButtonPress,
+      onFileButtonPress,
+      onLocationButtonPress,
+      onContactButtonPress,
+      onTakePhotoButtonPress,
+      onGalleryItemPress,
+    } = this.props;
+
+    return fileSectionOpened ? (
```

Comment:
I really don't like this. Is it better than doing null check in calling function?